src/main/webapp/js/addrole.js

```diff
@@ -93,6 +93,10 @@ const sendPostRequest = (postUrl, json) => {
     const xhr = new XMLHttpRequest();
     xhr.open('POST', postUrl, true);
     xhr.setRequestHeader('Content-Type', 'application/json');
+    // Jelly file sets up the crumb value for CSRF protection
+    if (crumb.value) {
```

Comment:
The `crumb` object is created by https://github.com/jenkinsci/jenkins/blob/dedde6195362b4aeae75f8a147cdf263b2f9854b/core/src/main/resources/lib/layout/html.jelly